Update sort-array.md

mrcdk opened the pull request:
Added an example using haxe.ds.ArraySort and http://try.haxe.org/#56568 to show the differences between both array.sort() and haxe.ds.ArraySort.sort()

I'm not sure if we should keep the reflection example.

markknol commented:
`[tryhaxe](http://try.haxe.org/#56568)` should be `[tryhaxe](http://try.haxe.org/embed/56568)`

Simn commented:
This should also mention that ArraySort.sort is stable and Array.sort is not (necessarily, that might depend on the target). We could even link to https://en.wikipedia.org/wiki/Sorting_algorithm#Stability or something.

mrcdk commented:
Okay, I've updated it with the link @Simn posted. I've also removed the reflection example because I feel it should be in another place. Maybe some other entry where `Reflect` is explained.

Simn commented:
Very nice, thanks!
